src/app/components/dialog/dialog.ts

```diff
@@ -171,6 +188,8 @@ export class Dialog implements AfterContentInit,OnInit,OnDestroy {
 
     @Input() maximizeIcon: string = 'pi pi-window-maximize';
 
+    @Input() headerButtons: any[] = null;
```

Comment:
mb use `ng-content` for additional buttons inside `p-dialog-header-icons` instead of `@Input`?

Reply:
I think it would be more flexible, but then you could also add other elements which could break the structure and formatting of the header. In my opinion it is better when you only have the option to add custom buttons to the top right part of the dialog as it is intended by the framework.